modules/apps/site/site-cms-site-initializer/src/main/java/com/liferay/site/cms/site/initializer/internal/display/context/FilesSectionDisplayContext.java

```diff
@@ -130,6 +130,16 @@ public List<FDSActionDropdownItem> getFDSActionDropdownItems() {
 				"{embedded.file.link.href}", "download", "download",
 				LanguageUtil.get(httpServletRequest, "download"), "get", null,
 				"link"));
+		fdsActionDropdownItems.add(
+			3,
+			new FDSActionDropdownItem(
+				StringBundler.concat(
+					"/o", GroupConstants.CMS_FRIENDLY_URL, "download-folder/",
+					_portal.getClassNameId(ObjectEntryFolder.class),
+					"/{embedded.id}"),
+				"download", "download-folder",
+				LanguageUtil.get(httpServletRequest, "download"), "get", null,
+				"link"));
```

Comment:
You can add a "conditional" here to show or not the action, like Marco did in https://github.com/liferay-content-management/liferay-portal/pull/6677/commits/5b26334993481b750684b51487248667e01a5314. 

This way, no change in the FilesFDSPropsTransformer is needed, and everything is more clear.